Dockerfile

```diff
@@ -1,22 +1,44 @@
-FROM ruby:2.7.5
-RUN apt-get update -qq && apt-get upgrade -y && apt-get install -y build-essential && apt-get clean
-RUN gem install foreman
+ARG base_image=ruby:2.7.5-slim-buster
 
-ENV GOVUK_APP_NAME router-api
-ENV MONGODB_URI mongodb://mongo/router
-ENV PORT 3056
-ENV RAILS_ENV development
-ENV TEST_MONGODB_URI mongodb://mongo/router-test
+FROM $base_image AS builder
+
+ENV RAILS_ENV=production
+
+# TODO: have a separate build image which already contains the build-only deps.
+RUN apt-get update -qq && \
+    apt-get upgrade -y && \
+    apt-get install -y build-essential nodejs wget && \
+    apt-get clean
+
+RUN mkdir /app
+
+WORKDIR /app
+COPY Gemfile* .ruby-version /app/
+
+RUN bundle config set deployment 'true' && \
+    bundle config set without 'development test webkit' && \
+    bundle install -j8 --retry=2
 
-# place the AWS RDS Certificate Authority bundle at well known path
 RUN wget -O /etc/ssl/certs/rds-combined-ca-bundle.pem https://s3.amazonaws.com/rds-downloads/rds-combined-ca-bundle.pem
```

Comment:
Maybe better to just have this Run done in the final build stage rather then here

Reply:
fair enough thank you